tests/smoke/base.sh

```diff
@@ -49,17 +49,21 @@ elif command -v meson &> /dev/null; then
   pip install -e .
 elif command -v pdm &> /dev/null; then
   pdm install
+elif command -v hatch &> /dev/null; then
+  COMMAND_PREFIX="hatch run"
```

Comment:
I think this should still be `pip install --editable .`. We do need to check if the package is being installed correctly. The current passing of CI might be a false positive.

Reply:
@Saransh-cpp I tried using `pip install --editable .` but it didn't work out. @xmnlab  has suggested me to use this method for installing hatch.

Reply:
That sounds weird 🙂 

I install all my hatch-based packages with `pip install .` and it works fine for me. We might want to look at this in the future.